core/src/main/scala/org/json4s/Extraction.scala

```diff
@@ -609,7 +609,18 @@ object Extraction {
         case other: JValue => other
       }
 
-      val args = constructor.params.map(a => buildCtorArg(deserializedJson \ a.name, a))
+      val paramsWithOriginalTypes = constructor.params.zip(jconstructor.getParameterTypes())
+
+      val args = paramsWithOriginalTypes.collect {
+        case (param, clazz) =>
+          val rawArg = buildCtorArg(deserializedJson \ param.name, param)
+          (rawArg, clazz) match {
+            case (arg:BigDecimal,tt) if tt == classOf[java.math.BigDecimal] => arg.bigDecimal
```

Comment:
Can be `classOf[JavaBigDecimal]` because the class has `import java.math.{BigDecimal => JavaBigDecimal}`.

Reply:
Having a space after `:` like s/arg:BigDecimal/arg: BigDecimal/ is consistent with current code. Same for the test code.